assets/text/message_data.h

```diff
@@ -0,0 +1 @@
+#include "version_assets/text/message_data.h"
```

Comment:
Oh I see, this was the purpose behind `version_assets`. Does this file need to exist at all if the include paths are set up correctly?

Reply:
Having an indirection like this is very useful for modding, it means you can add your own messages directly in git-tracked assets/text/message_data.h, and can remove the vanilla messages just by not including them. Or using messages from a specific version with `#include "assets/.../gc-eu-mq/.../message_data.h"`

This may seem a bit contrived in the case of text, but this approach is generalizing what I want to do for all assets, that is committing .c files that #include extracted assets, which has the same benefits as listed above.

as to version_assets/ yeah I think it is clearer with it where it's pulling from, even if not technically required. but I'm open to changing it (I do value a lot ability to understand what's going on)

Reply:
I think it would be pretty understandable if the assets were only built out of `assets/$(VERSION)`, the extra indirection is more complex.

Could we revisit the modding issue later? With no indirection, this PR seems compatible with the current modding workflow, which IIUC is tracking custom assets in a separate folder and cp-ing them into `assets/` during the build. Perhaps later if we deem a `version_assets` path component is necessary we can change it then

Reply:
> Could we revisit the modding issue later?

no, that's the main thing this build system change addresses

Reply:
Oh, I thought the primary reason was to support multiversion assets for building matching ROMs. Could we separate the multiversion assets vs modding issues then, and make a separate PR for versioning the assets dir?

Reply:
I don't see the point given both involve directory structure layout, may as well resolve both at once rather than artificially one after the other

Reply:
Because I agree with the multiversion assets approach (split asset directories by version) but not yet with the modding approach (layer of indirection with #include). So splitting them is a way for us to make progress without stalling everything

Reply:
> not yet with the modding approach (layer of indirection with #include)

I know you weren't involved then but I made that suggestion over a year ago
https://discord.com/channels/688807550715560050/688851451014938642/1051488155099987998
https://github.com/Dragorn421/decomp_assets_and_multiversion_demo
and no other option has emerged, this option has even gained some implicit approvals